src/cargo/core/compiler/custom_build.rs

```diff
@@ -249,6 +249,28 @@ fn build_work(cx: &mut Context<'_, '_>, unit: &Unit) -> CargoResult<Job> {
         }
     }
 
+    // Also inform the build script of the rustc compiler context.
+    if let Some(wrapper) = bcx.rustc().wrapper.as_ref() {
+        cmd.env("RUSTC_WRAPPER", wrapper);
+    }
+    cmd.env("RUSTFLAGS", bcx.rustflags_args(unit).join(" "));
+    let version = crate::version();
+    cmd.env(
+        "CARGO_VERSION",
+        format!("{}.{}.{}", version.major, version.minor, version.patch),
+    );
+    cmd.env("CARGO_VERSION_MAJOR", version.major.to_string());
+    cmd.env("CARGO_VERSION_MINOR", version.minor.to_string());
+    cmd.env("CARGO_VERSION_PATCH", version.patch.to_string());
```

Comment:
On second though, Cargo's version is implied by or should always match rustc's version, so I think it's fine to avoid exporting these env vars?

Reply:
But that may not be true when the likes of gcc-rs come online -- unless you expect them to masquerade, user-agent style?

Reply:
I mean I don't think anyone wants user-agent madness or anything like that. Cargo is a long way from supporting an alternate compiler, and if necessary we could add the cargo version at that point, but for now I don't want to proliferate a possible misunderstanding that cargo/rustc are intended to be separately versioned. They're currently joined at the hip and are distributed together, and if we ever change that a lot more needs to change than this one env var.

Reply:
Pushed a commit that removes `CARGO_VERSION*`.